### ISSUE TITLE
Databricks: Missing Syntax Support For `?::` as a Shorthand For `TRY_CAST`

### ISSUE THREAD

**Issue by beepboopcollins** [sqlfluff/sqlfluff#7926]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Unable to parse statements containing the "question double colon" (`?::`) shorthand syntax for `try_cast` using databricks dialect.

### Expected Behaviour

Section is parsed and no error is raised.

### Observed Behaviour

When parsing 

```
SELECT 1?::INT
```

as dialect databricks

We get 

L:   1 | P:   9 |  PRS | Line 1, Position 9: Found unparsable section: '?::INT'

Even though this is valid syntax in databricks as of version 15.3 and above

https://docs.databricks.com/aws/en/sql/language-manual/functions/questiondoublecolonsign

### How to reproduce

Create a file called "test.sql" containing this text:

```
SELECT 1?::INT
```

run `sqlfluff lint --dialect databricks`

observe 
L:   1 | P:   9 |  PRS | Line 1, Position 9: Found unparsable section: '?::INT'

### Dialect

Databricks 

### Version

sqlfluff, version 4.2.2

### Configuration

Empty

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
